# Test model yolov10x

In [1]:
!git clone https://github.com/THU-MIG/yolov10.git

%cd yolov10

!pip install .

from ultralytics import YOLOv10

Cloning into 'yolov10'...
remote: Enumerating objects: 20291, done.
remote: Counting objects: 100% (370/370), done.
remote: Compressing objects: 100% (174/174), done.
remote: Total 20291 (delta 249), reused 303 (delta 196), pack-reused 19921
Receiving objects: 100% (20291/20291), 11.37 MiB | 24.76 MiB/s, done.
Resolving deltas: 100% (14228/14228), done.
/kaggle/working/yolov10
Processing /kaggle/working/yolov10
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
  Created wheel for ultralytics: filename=ultralytics-8.1.34-py3-none-any.whl size=731149 sha256=e1ffd15e9965cfc7ee744ef5835fadf53934f4833a74ca3320c0d5d93815bd2d
  Stored in directory: /tmp/pip-ephem-wheel-cache-3sy90fdt/wheels/42/8a/a9/9cce36b0b3d4ada7292a5a495abef6993ad1db52a921496752
Successfully built ultralytics


**fold0**

In [2]:
#test yolov10x fold0
model1 = YOLOv10('/kaggle/input/test010/yolov10x_100epochs_fold0_best.pt')
model1.val(data='/kaggle/input/detecting-student-classroom-behavior2/kaggle_config_test.yaml')

Ultralytics YOLOv8.1.34 🚀 Python-3.10.13 torch-2.1.2 CUDA:0 (Tesla T4, 15102MiB)
YOLOv10x summary (fused): 503 layers, 31589858 parameters, 0 gradients, 169.8 GFLOPs


100%|██████████| 755k/755k [00:00<00:00, 14.3MB/s]
val: Scanning /kaggle/input/detecting-student-classroom-behavior2/data_test2/data_test2/labels/test... 1026 images, 0 backgrounds, 0 corrupt: 100%|██████████| 1026/1026 [00:07<00:00, 128.58it/s]

val: WARNING ⚠️ /kaggle/input/detecting-student-classroom-behavior2/data_test2/data_test2/images/test/3005055.jpg: 1 duplicate labels removed


val: WARNING ⚠️ Cache directory /kaggle/input/detecting-student-classroom-behavior2/data_test2/data_test2/labels is not writeable, cache not saved.


                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 65/65 [00:46<00:00,  1.40it/s]


                   all       1026       5426      0.705      0.736      0.763      0.586
          hand-raising       1026       2310      0.805      0.795      0.849      0.635
               reading       1026       2433      0.674      0.756      0.773      0.605
               writing       1026        683      0.638      0.657      0.666      0.518
Speed: 0.2ms preprocess, 37.3ms inference, 0.0ms loss, 0.4ms postprocess per image
Results saved to /kaggle/working/yolov10/runs/detect/val


ultralytics.utils.metrics.DetMetrics object with attributes:

ap_class_index: array([0, 1, 2])
box: ultralytics.utils.metrics.Metric object
confusion_matrix: <ultralytics.utils.metrics.ConfusionMatrix object at 0x7ffac0484580>
curves: ['Precision-Recall(B)', 'F1-Confidence(B)', 'Precision-Confidence(B)', 'Recall-Confidence(B)']
curves_results: [[array([          0,    0.001001,    0.002002,    0.003003,    0.004004,    0.005005,    0.006006,    0.007007,    0.008008,    0.009009,     0.01001,    0.011011,    0.012012,    0.013013,    0.014014,    0.015015,    0.016016,    0.017017,    0.018018,    0.019019,     0.02002,    0.021021,    0.022022,    0.023023,
          0.024024,    0.025025,    0.026026,    0.027027,    0.028028,    0.029029,     0.03003,    0.031031,    0.032032,    0.033033,    0.034034,    0.035035,    0.036036,    0.037037,    0.038038,    0.039039,     0.04004,    0.041041,    0.042042,    0.043043,    0.044044,    0.045045,    0.046046,    0.047047,
          0.04

**fold2**

In [3]:
#test yolov10x fold2
model2 = YOLOv10('/kaggle/input/test010/yolov10x_100epochs_fold2_best.pt')
model2.val(data='/kaggle/input/detecting-student-classroom-behavior2/kaggle_config_test.yaml')

Ultralytics YOLOv8.1.34 🚀 Python-3.10.13 torch-2.1.2 CUDA:0 (Tesla T4, 15102MiB)
YOLOv10x summary (fused): 503 layers, 31589858 parameters, 0 gradients, 169.8 GFLOPs


val: Scanning /kaggle/input/detecting-student-classroom-behavior2/data_test2/data_test2/labels/test... 1026 images, 0 backgrounds, 0 corrupt: 100%|██████████| 1026/1026 [00:02<00:00, 495.84it/s]

val: WARNING ⚠️ /kaggle/input/detecting-student-classroom-behavior2/data_test2/data_test2/images/test/3005055.jpg: 1 duplicate labels removed


val: WARNING ⚠️ Cache directory /kaggle/input/detecting-student-classroom-behavior2/data_test2/data_test2/labels is not writeable, cache not saved.


                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 65/65 [00:46<00:00,  1.39it/s]


                   all       1026       5426      0.704      0.719      0.753      0.581
          hand-raising       1026       2310      0.816      0.775      0.845      0.635
               reading       1026       2433      0.668      0.739      0.751      0.587
               writing       1026        683      0.629      0.642      0.662      0.521
Speed: 0.2ms preprocess, 37.2ms inference, 0.0ms loss, 0.5ms postprocess per image
Results saved to /kaggle/working/yolov10/runs/detect/val2


ultralytics.utils.metrics.DetMetrics object with attributes:

ap_class_index: array([0, 1, 2])
box: ultralytics.utils.metrics.Metric object
confusion_matrix: <ultralytics.utils.metrics.ConfusionMatrix object at 0x7ff87aae2bc0>
curves: ['Precision-Recall(B)', 'F1-Confidence(B)', 'Precision-Confidence(B)', 'Recall-Confidence(B)']
curves_results: [[array([          0,    0.001001,    0.002002,    0.003003,    0.004004,    0.005005,    0.006006,    0.007007,    0.008008,    0.009009,     0.01001,    0.011011,    0.012012,    0.013013,    0.014014,    0.015015,    0.016016,    0.017017,    0.018018,    0.019019,     0.02002,    0.021021,    0.022022,    0.023023,
          0.024024,    0.025025,    0.026026,    0.027027,    0.028028,    0.029029,     0.03003,    0.031031,    0.032032,    0.033033,    0.034034,    0.035035,    0.036036,    0.037037,    0.038038,    0.039039,     0.04004,    0.041041,    0.042042,    0.043043,    0.044044,    0.045045,    0.046046,    0.047047,
          0.04

**fold3**

In [4]:
#test yolov10x fold3
model3 = YOLOv10('/kaggle/input/test010/yolov10x_100epochs_fold3_best.pt')
model3.val(data='/kaggle/input/detecting-student-classroom-behavior2/kaggle_config_test.yaml')

Ultralytics YOLOv8.1.34 🚀 Python-3.10.13 torch-2.1.2 CUDA:0 (Tesla T4, 15102MiB)
YOLOv10x summary (fused): 503 layers, 31589858 parameters, 0 gradients, 169.8 GFLOPs


val: Scanning /kaggle/input/detecting-student-classroom-behavior2/data_test2/data_test2/labels/test... 1026 images, 0 backgrounds, 0 corrupt: 100%|██████████| 1026/1026 [00:01<00:00, 562.70it/s]

val: WARNING ⚠️ /kaggle/input/detecting-student-classroom-behavior2/data_test2/data_test2/images/test/3005055.jpg: 1 duplicate labels removed


val: WARNING ⚠️ Cache directory /kaggle/input/detecting-student-classroom-behavior2/data_test2/data_test2/labels is not writeable, cache not saved.


                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 65/65 [00:47<00:00,  1.37it/s]


                   all       1026       5426       0.74      0.691      0.756      0.585
          hand-raising       1026       2310      0.828      0.784      0.851      0.642
               reading       1026       2433      0.718      0.703      0.763      0.597
               writing       1026        683      0.674      0.587      0.654      0.515
Speed: 0.2ms preprocess, 38.0ms inference, 0.0ms loss, 0.5ms postprocess per image
Results saved to /kaggle/working/yolov10/runs/detect/val3


ultralytics.utils.metrics.DetMetrics object with attributes:

ap_class_index: array([0, 1, 2])
box: ultralytics.utils.metrics.Metric object
confusion_matrix: <ultralytics.utils.metrics.ConfusionMatrix object at 0x7ff879c98ee0>
curves: ['Precision-Recall(B)', 'F1-Confidence(B)', 'Precision-Confidence(B)', 'Recall-Confidence(B)']
curves_results: [[array([          0,    0.001001,    0.002002,    0.003003,    0.004004,    0.005005,    0.006006,    0.007007,    0.008008,    0.009009,     0.01001,    0.011011,    0.012012,    0.013013,    0.014014,    0.015015,    0.016016,    0.017017,    0.018018,    0.019019,     0.02002,    0.021021,    0.022022,    0.023023,
          0.024024,    0.025025,    0.026026,    0.027027,    0.028028,    0.029029,     0.03003,    0.031031,    0.032032,    0.033033,    0.034034,    0.035035,    0.036036,    0.037037,    0.038038,    0.039039,     0.04004,    0.041041,    0.042042,    0.043043,    0.044044,    0.045045,    0.046046,    0.047047,
          0.04

# Test model yolov9e

In [5]:
!pip install ultralytics

In [6]:
pip install -U ipywidgets

   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 139.4/139.4 kB 1.4 MB/s eta 0:00:00 0:00:010m
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 214.4/214.4 kB 3.7 MB/s eta 0:00:00a 0:00:01
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 2.3/2.3 MB 22.9 MB/s eta 0:00:00a 0:00:01
  Attempting uninstall: widgetsnbextension
    Found existing installation: widgetsnbextension 3.6.6
    Uninstalling widgetsnbextension-3.6.6:
      Successfully uninstalled widgetsnbextension-3.6.6
  Attempting uninstall: jupyterlab-widgets
    Found existing installation: jupyterlab-widgets 3.0.9
    Uninstalling jupyterlab-widgets-3.0.9:
      Successfully uninstalled jupyterlab-widgets-3.0.9
  Attempting uninstall: ipywidgets
    Found existing installation: ipywidgets 7.7.1
    Uninstalling ipywidgets-7.7.1:
      Successfully uninstalled ipywidgets-7.7.1
Note: you may need to restart the kernel to use updated packages.


In [7]:
from ultralytics import YOLO

**fold0**

In [8]:
#test yolov9e fold0
model4 = YOLO('/kaggle/input/test010/yolov9e_100epochs_fold0_best.pt')
model4.val(data="/kaggle/input/detecting-student-classroom-behavior2/kaggle_config_test.yaml")

Ultralytics YOLOv8.1.34 🚀 Python-3.10.13 torch-2.1.2 CUDA:0 (Tesla T4, 15102MiB)
YOLOv9e summary (fused): 687 layers, 57378713 parameters, 0 gradients, 189.1 GFLOPs


val: Scanning /kaggle/input/detecting-student-classroom-behavior2/data_test2/data_test2/labels/test... 1026 images, 0 backgrounds, 0 corrupt: 100%|██████████| 1026/1026 [00:01<00:00, 568.74it/s]

val: WARNING ⚠️ /kaggle/input/detecting-student-classroom-behavior2/data_test2/data_test2/images/test/3005055.jpg: 1 duplicate labels removed
val: WARNING ⚠️ Cache directory /kaggle/input/detecting-student-classroom-behavior2/data_test2/data_test2/labels is not writeable, cache not saved.



                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 65/65 [00:52<00:00,  1.23it/s]


                   all       1026       5426      0.701      0.746      0.776      0.605
          hand-raising       1026       2310       0.79      0.809      0.856      0.652
               reading       1026       2433      0.681       0.76      0.782       0.62
               writing       1026        683      0.633      0.669      0.691      0.545
Speed: 0.3ms preprocess, 44.6ms inference, 0.0ms loss, 3.0ms postprocess per image
Results saved to /kaggle/working/yolov10/runs/detect/val4


ultralytics.utils.metrics.DetMetrics object with attributes:

ap_class_index: array([0, 1, 2])
box: ultralytics.utils.metrics.Metric object
confusion_matrix: <ultralytics.utils.metrics.ConfusionMatrix object at 0x7ff8700e3160>
curves: ['Precision-Recall(B)', 'F1-Confidence(B)', 'Precision-Confidence(B)', 'Recall-Confidence(B)']
curves_results: [[array([          0,    0.001001,    0.002002,    0.003003,    0.004004,    0.005005,    0.006006,    0.007007,    0.008008,    0.009009,     0.01001,    0.011011,    0.012012,    0.013013,    0.014014,    0.015015,    0.016016,    0.017017,    0.018018,    0.019019,     0.02002,    0.021021,    0.022022,    0.023023,
          0.024024,    0.025025,    0.026026,    0.027027,    0.028028,    0.029029,     0.03003,    0.031031,    0.032032,    0.033033,    0.034034,    0.035035,    0.036036,    0.037037,    0.038038,    0.039039,     0.04004,    0.041041,    0.042042,    0.043043,    0.044044,    0.045045,    0.046046,    0.047047,
          0.04

**fold2**

In [9]:
#test yolov9e fold2
model5 = YOLO('/kaggle/input/test010/yolov9e_100epochs_fold2_best.pt')
model5.val(data="/kaggle/input/detecting-student-classroom-behavior2/kaggle_config_test.yaml")

Ultralytics YOLOv8.1.34 🚀 Python-3.10.13 torch-2.1.2 CUDA:0 (Tesla T4, 15102MiB)
YOLOv9e summary (fused): 687 layers, 57378713 parameters, 0 gradients, 189.1 GFLOPs


val: Scanning /kaggle/input/detecting-student-classroom-behavior2/data_test2/data_test2/labels/test... 1026 images, 0 backgrounds, 0 corrupt: 100%|██████████| 1026/1026 [00:01<00:00, 589.10it/s]

val: WARNING ⚠️ /kaggle/input/detecting-student-classroom-behavior2/data_test2/data_test2/images/test/3005055.jpg: 1 duplicate labels removed


val: WARNING ⚠️ Cache directory /kaggle/input/detecting-student-classroom-behavior2/data_test2/data_test2/labels is not writeable, cache not saved.


                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 65/65 [00:51<00:00,  1.26it/s]


                   all       1026       5426      0.752      0.705      0.774      0.611
          hand-raising       1026       2310      0.843      0.777      0.861      0.659
               reading       1026       2433      0.731      0.716       0.78      0.625
               writing       1026        683      0.681      0.622      0.682      0.548
Speed: 0.2ms preprocess, 45.4ms inference, 0.0ms loss, 0.9ms postprocess per image
Results saved to /kaggle/working/yolov10/runs/detect/val5


ultralytics.utils.metrics.DetMetrics object with attributes:

ap_class_index: array([0, 1, 2])
box: ultralytics.utils.metrics.Metric object
confusion_matrix: <ultralytics.utils.metrics.ConfusionMatrix object at 0x7ff6ca443610>
curves: ['Precision-Recall(B)', 'F1-Confidence(B)', 'Precision-Confidence(B)', 'Recall-Confidence(B)']
curves_results: [[array([          0,    0.001001,    0.002002,    0.003003,    0.004004,    0.005005,    0.006006,    0.007007,    0.008008,    0.009009,     0.01001,    0.011011,    0.012012,    0.013013,    0.014014,    0.015015,    0.016016,    0.017017,    0.018018,    0.019019,     0.02002,    0.021021,    0.022022,    0.023023,
          0.024024,    0.025025,    0.026026,    0.027027,    0.028028,    0.029029,     0.03003,    0.031031,    0.032032,    0.033033,    0.034034,    0.035035,    0.036036,    0.037037,    0.038038,    0.039039,     0.04004,    0.041041,    0.042042,    0.043043,    0.044044,    0.045045,    0.046046,    0.047047,
          0.04

**fold3**

In [10]:
#test yolov9e fold3
model6 = YOLO('/kaggle/input/test010/yolov9e_100epochs_fold3_best.pt')
model6.val(data="/kaggle/input/detecting-student-classroom-behavior2/kaggle_config_test.yaml")

Ultralytics YOLOv8.1.34 🚀 Python-3.10.13 torch-2.1.2 CUDA:0 (Tesla T4, 15102MiB)
YOLOv9e summary (fused): 687 layers, 57378713 parameters, 0 gradients, 189.1 GFLOPs


val: Scanning /kaggle/input/detecting-student-classroom-behavior2/data_test2/data_test2/labels/test... 1026 images, 0 backgrounds, 0 corrupt: 100%|██████████| 1026/1026 [00:01<00:00, 564.72it/s]

val: WARNING ⚠️ /kaggle/input/detecting-student-classroom-behavior2/data_test2/data_test2/images/test/3005055.jpg: 1 duplicate labels removed


val: WARNING ⚠️ Cache directory /kaggle/input/detecting-student-classroom-behavior2/data_test2/data_test2/labels is not writeable, cache not saved.


                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 65/65 [00:51<00:00,  1.26it/s]


                   all       1026       5426      0.722       0.73      0.778      0.611
          hand-raising       1026       2310      0.808      0.795      0.854      0.658
               reading       1026       2433      0.696      0.756      0.783      0.626
               writing       1026        683      0.661       0.64      0.698      0.549
Speed: 0.2ms preprocess, 45.3ms inference, 0.0ms loss, 1.2ms postprocess per image
Results saved to /kaggle/working/yolov10/runs/detect/val6


ultralytics.utils.metrics.DetMetrics object with attributes:

ap_class_index: array([0, 1, 2])
box: ultralytics.utils.metrics.Metric object
confusion_matrix: <ultralytics.utils.metrics.ConfusionMatrix object at 0x7ff6c8653490>
curves: ['Precision-Recall(B)', 'F1-Confidence(B)', 'Precision-Confidence(B)', 'Recall-Confidence(B)']
curves_results: [[array([          0,    0.001001,    0.002002,    0.003003,    0.004004,    0.005005,    0.006006,    0.007007,    0.008008,    0.009009,     0.01001,    0.011011,    0.012012,    0.013013,    0.014014,    0.015015,    0.016016,    0.017017,    0.018018,    0.019019,     0.02002,    0.021021,    0.022022,    0.023023,
          0.024024,    0.025025,    0.026026,    0.027027,    0.028028,    0.029029,     0.03003,    0.031031,    0.032032,    0.033033,    0.034034,    0.035035,    0.036036,    0.037037,    0.038038,    0.039039,     0.04004,    0.041041,    0.042042,    0.043043,    0.044044,    0.045045,    0.046046,    0.047047,
          0.04

# Test model yolov8x

**fold0**

In [11]:
#test yolov8x fold0
model7 = YOLO('/kaggle/input/test010/yolov8x_100epochs_fold0_best.pt')
model7.val(data="/kaggle/input/detecting-student-classroom-behavior2/kaggle_config_test.yaml")

Ultralytics YOLOv8.1.34 🚀 Python-3.10.13 torch-2.1.2 CUDA:0 (Tesla T4, 15102MiB)
Model summary (fused): 268 layers, 68126457 parameters, 0 gradients, 257.4 GFLOPs


val: Scanning /kaggle/input/detecting-student-classroom-behavior2/data_test2/data_test2/labels/test... 1026 images, 0 backgrounds, 0 corrupt: 100%|██████████| 1026/1026 [00:01<00:00, 548.66it/s]

val: WARNING ⚠️ /kaggle/input/detecting-student-classroom-behavior2/data_test2/data_test2/images/test/3005055.jpg: 1 duplicate labels removed
val: WARNING ⚠️ Cache directory /kaggle/input/detecting-student-classroom-behavior2/data_test2/data_test2/labels is not writeable, cache not saved.



                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 65/65 [00:46<00:00,  1.40it/s]


                   all       1026       5426      0.709      0.725      0.763      0.595
          hand-raising       1026       2310      0.829      0.791      0.853      0.643
               reading       1026       2433      0.696      0.736      0.773      0.614
               writing       1026        683      0.601      0.649      0.662      0.528
Speed: 0.2ms preprocess, 37.0ms inference, 0.0ms loss, 4.1ms postprocess per image
Results saved to /kaggle/working/yolov10/runs/detect/val7


ultralytics.utils.metrics.DetMetrics object with attributes:

ap_class_index: array([0, 1, 2])
box: ultralytics.utils.metrics.Metric object
confusion_matrix: <ultralytics.utils.metrics.ConfusionMatrix object at 0x7ff6ba4ce9b0>
curves: ['Precision-Recall(B)', 'F1-Confidence(B)', 'Precision-Confidence(B)', 'Recall-Confidence(B)']
curves_results: [[array([          0,    0.001001,    0.002002,    0.003003,    0.004004,    0.005005,    0.006006,    0.007007,    0.008008,    0.009009,     0.01001,    0.011011,    0.012012,    0.013013,    0.014014,    0.015015,    0.016016,    0.017017,    0.018018,    0.019019,     0.02002,    0.021021,    0.022022,    0.023023,
          0.024024,    0.025025,    0.026026,    0.027027,    0.028028,    0.029029,     0.03003,    0.031031,    0.032032,    0.033033,    0.034034,    0.035035,    0.036036,    0.037037,    0.038038,    0.039039,     0.04004,    0.041041,    0.042042,    0.043043,    0.044044,    0.045045,    0.046046,    0.047047,
          0.04

**fold2**

In [12]:
#test yolov8x fold2
model8 = YOLO('/kaggle/input/test010/yolov8x_100epochs_fold2_best.pt')
model8.val(data="/kaggle/input/detecting-student-classroom-behavior2/kaggle_config_test.yaml")

Ultralytics YOLOv8.1.34 🚀 Python-3.10.13 torch-2.1.2 CUDA:0 (Tesla T4, 15102MiB)
Model summary (fused): 268 layers, 68126457 parameters, 0 gradients, 257.4 GFLOPs


val: Scanning /kaggle/input/detecting-student-classroom-behavior2/data_test2/data_test2/labels/test... 1026 images, 0 backgrounds, 0 corrupt: 100%|██████████| 1026/1026 [00:01<00:00, 743.92it/s]

val: WARNING ⚠️ /kaggle/input/detecting-student-classroom-behavior2/data_test2/data_test2/images/test/3005055.jpg: 1 duplicate labels removed
val: WARNING ⚠️ Cache directory /kaggle/input/detecting-student-classroom-behavior2/data_test2/data_test2/labels is not writeable, cache not saved.



                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 65/65 [00:46<00:00,  1.39it/s]


                   all       1026       5426      0.722      0.733      0.764      0.599
          hand-raising       1026       2310      0.824      0.797      0.853      0.655
               reading       1026       2433      0.705      0.727      0.762      0.608
               writing       1026        683      0.638      0.676      0.676      0.535
Speed: 0.3ms preprocess, 36.2ms inference, 0.0ms loss, 3.3ms postprocess per image
Results saved to /kaggle/working/yolov10/runs/detect/val8


ultralytics.utils.metrics.DetMetrics object with attributes:

ap_class_index: array([0, 1, 2])
box: ultralytics.utils.metrics.Metric object
confusion_matrix: <ultralytics.utils.metrics.ConfusionMatrix object at 0x7ff6b820b670>
curves: ['Precision-Recall(B)', 'F1-Confidence(B)', 'Precision-Confidence(B)', 'Recall-Confidence(B)']
curves_results: [[array([          0,    0.001001,    0.002002,    0.003003,    0.004004,    0.005005,    0.006006,    0.007007,    0.008008,    0.009009,     0.01001,    0.011011,    0.012012,    0.013013,    0.014014,    0.015015,    0.016016,    0.017017,    0.018018,    0.019019,     0.02002,    0.021021,    0.022022,    0.023023,
          0.024024,    0.025025,    0.026026,    0.027027,    0.028028,    0.029029,     0.03003,    0.031031,    0.032032,    0.033033,    0.034034,    0.035035,    0.036036,    0.037037,    0.038038,    0.039039,     0.04004,    0.041041,    0.042042,    0.043043,    0.044044,    0.045045,    0.046046,    0.047047,
          0.04

**fold3**

In [13]:
#test yolov8x fold3
model9 = YOLO('/kaggle/input/test010/yolov8x_100epochs_fold3_best.pt')
model9.val(data="/kaggle/input/detecting-student-classroom-behavior2/kaggle_config_test.yaml")

Ultralytics YOLOv8.1.34 🚀 Python-3.10.13 torch-2.1.2 CUDA:0 (Tesla T4, 15102MiB)
Model summary (fused): 268 layers, 68126457 parameters, 0 gradients, 257.4 GFLOPs


val: Scanning /kaggle/input/detecting-student-classroom-behavior2/data_test2/data_test2/labels/test... 1026 images, 0 backgrounds, 0 corrupt: 100%|██████████| 1026/1026 [00:01<00:00, 712.84it/s]

val: WARNING ⚠️ /kaggle/input/detecting-student-classroom-behavior2/data_test2/data_test2/images/test/3005055.jpg: 1 duplicate labels removed


val: WARNING ⚠️ Cache directory /kaggle/input/detecting-student-classroom-behavior2/data_test2/data_test2/labels is not writeable, cache not saved.


                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 65/65 [00:47<00:00,  1.38it/s]


                   all       1026       5426      0.744      0.693      0.754      0.598
          hand-raising       1026       2310      0.848      0.782      0.856      0.665
               reading       1026       2433      0.733      0.694      0.752      0.602
               writing       1026        683      0.651      0.603      0.653      0.528
Speed: 0.3ms preprocess, 36.1ms inference, 0.0ms loss, 2.4ms postprocess per image
Results saved to /kaggle/working/yolov10/runs/detect/val9


ultralytics.utils.metrics.DetMetrics object with attributes:

ap_class_index: array([0, 1, 2])
box: ultralytics.utils.metrics.Metric object
confusion_matrix: <ultralytics.utils.metrics.ConfusionMatrix object at 0x7ff6b6432860>
curves: ['Precision-Recall(B)', 'F1-Confidence(B)', 'Precision-Confidence(B)', 'Recall-Confidence(B)']
curves_results: [[array([          0,    0.001001,    0.002002,    0.003003,    0.004004,    0.005005,    0.006006,    0.007007,    0.008008,    0.009009,     0.01001,    0.011011,    0.012012,    0.013013,    0.014014,    0.015015,    0.016016,    0.017017,    0.018018,    0.019019,     0.02002,    0.021021,    0.022022,    0.023023,
          0.024024,    0.025025,    0.026026,    0.027027,    0.028028,    0.029029,     0.03003,    0.031031,    0.032032,    0.033033,    0.034034,    0.035035,    0.036036,    0.037037,    0.038038,    0.039039,     0.04004,    0.041041,    0.042042,    0.043043,    0.044044,    0.045045,    0.046046,    0.047047,
          0.04

# Summary

 **yolov10x**
 
 fold0
* 					Class     Images  Instances      Box(P          R      mAP50  mAP50-95):
*                    all       1026       5426      0.705      0.736      0.763      0.586
*           hand-raising       1026       2310      0.805      0.795      0.849      0.635
*                reading       1026       2433      0.674      0.756      0.773      0.605
*                writing       1026        683      0.638      0.657      0.666      0.518
 
 fold2
* 					Class     Images  Instances      Box(P          R      mAP50  mAP50-95):
*                    all       1026       5426      0.704      0.719      0.753      0.581
*           hand-raising       1026       2310      0.816      0.775      0.845      0.635
*                reading       1026       2433      0.668      0.739      0.751      0.587
*                writing       1026        683      0.629      0.642      0.662      0.521
 
 fold3
* 					Class     Images  Instances      Box(P          R      mAP50  mAP50-95):
*                    all       1026       5426       0.74      0.691      0.756      0.585
*           hand-raising       1026       2310      0.828      0.784      0.851      0.642
*                reading       1026       2433      0.718      0.703      0.763      0.597
*                writing       1026        683      0.674      0.587      0.654      0.515
 
 
 
 **yolov9e**
 
 fold0
* 					Class     Images  Instances      Box(P          R      mAP50  mAP50-95):
*                    all       1026       5426      0.701      0.746      0.776      0.605
*           hand-raising       1026       2310       0.79      0.809      0.856      0.652
*                reading       1026       2433      0.681       0.76      0.782       0.62
*                writing       1026        683      0.633      0.669      0.691      0.545
 
 fold2
* 					Class     Images  Instances      Box(P          R      mAP50  mAP50-95):
*                    all       1026       5426      0.752      0.705      0.774      0.611
*           hand-raising       1026       2310      0.843      0.777      0.861      0.659
*                reading       1026       2433      0.731      0.716       0.78      0.625
*                writing       1026        683      0.681      0.622      0.682      0.548
 
 fold3
* 					Class     Images  Instances      Box(P          R      mAP50  mAP50-95):
*                    all       1026       5426      0.722       0.73      0.778      0.611
*           hand-raising       1026       2310      0.808      0.795      0.854      0.658
*                reading       1026       2433      0.696      0.756      0.783      0.626
*                writing       1026        683      0.661       0.64      0.698      0.549
 
 
 
 **yolov8x**
 
 fold0
* 					Class     Images  Instances      Box(P          R      mAP50  mAP50-95):
*                    all       1026       5426      0.709      0.725      0.763      0.595
*           hand-raising       1026       2310      0.829      0.791      0.853      0.643
*                reading       1026       2433      0.696      0.736      0.773      0.614
*                writing       1026        683      0.601      0.649      0.662      0.528
 
 fold2
* 					Class     Images  Instances      Box(P          R      mAP50  mAP50-95):
*                    all       1026       5426      0.722      0.733      0.764      0.599
*           hand-raising       1026       2310      0.824      0.797      0.853      0.655
*                reading       1026       2433      0.705      0.727      0.762      0.608
*                writing       1026        683      0.638      0.676      0.676      0.535

 fold3
* 					Class     Images  Instances      Box(P          R      mAP50  mAP50-95)
*                    all       1026       5426      0.744      0.693      0.754      0.598
*           hand-raising       1026       2310      0.848      0.782      0.856      0.665
*                reading       1026       2433      0.733      0.694      0.752      0.602
*                writing       1026        683      0.651      0.603      0.653      0.528

 **Best Model:**
* YOLOv9e shows consistently high mAP50 and mAP50-95 across all folds.
* YOLOv10x performs well but generally lower in mAP50-95 compared to YOLOv9e.

